src/app/@ansyn/ansyn/modules/plugins/cesium/plugins/annotations.visualizer.ts

```diff
@@ -0,0 +1,152 @@
+		mergeMap((isReady) => (isReady ? of(isReady) : this.isReady$.asObservable())),
+		take(1)
+	);
+
+	onAnnotationsChange$ = combineLatest(
+		this.store.pipe(select(selectLayersEntities)),
+		this.isAnnotationSubMenuOpen$,
+		this.store.select(selectSelectedLayersIds),
+		this.isActiveMap$,
+		this.store.select(selectActiveAnnotationLayer)
+	).pipe(mergeMap((resultArray) => this.renderEntities(...resultArray)));
+
+	@AutoSubscription
+	renderEntities$ = this.onCesiumDrawerReady$.pipe(switchMap(() => this.onAnnotationsChange$));
+
+	activeAnnotationLayer$: Observable<ILayer> = combineLatest(
+		this.store.pipe(select(selectActiveAnnotationLayer)),
+		this.store.pipe(select(selectLayersEntities))
+	).pipe(
+		map(([activeAnnotationLayerId, entities]) => {
+			return entities[activeAnnotationLayerId];
+		})
+	);
+
+	@AutoSubscription
+	onDrawEnd$ = () =>
+		this.cesiumDrawer.events.onDrawEnd.pipe(
+			withLatestFrom(this.activeAnnotationLayer$),
+			tap(([GeoJSON, activeAnnotationLayer]: [FeatureCollection<any>, ILayer]) => {
+				const data = <FeatureCollection<any>>{
+					...activeAnnotationLayer.data,
+					features: activeAnnotationLayer.data.features.concat(GeoJSON.features),
+				};
+				// TODO - display overlays with cesium
+				// if (this.overlay) {
+				// 	GeoJSON.features[0].properties = {
+				// 		...GeoJSON.features[0].properties,
+				// 		...this.projectionService.getProjectionProperties(this.communicator, data, feature, this.overlay)
+				// 	};
+				// }
+				this.store.dispatch(
+					new UpdateLayer({
+						id: activeAnnotationLayer.id,
+						data,
+					})
+				);
+			})
+		);
+
+	constructor(protected actions: Actions, public loggerService: LoggerService, public store: Store<any>) {
+		super();
+	}
+
+	renderEntities(
+		entities: {
+			[key: string]: ILayer;
+		},
+		isAnnotationSubMenuOpen: boolean,
+		selectedLayersIds: string[],
+		isActiveMap: boolean,
+		activeAnnotationLayer: string
+	): Observable<boolean> {
+		const displayedIds = uniq(
+			isActiveMap && isAnnotationSubMenuOpen ? [...selectedLayersIds, activeAnnotationLayer] : [...selectedLayersIds]
+		).filter((id: string) => entities[id] && entities[id].type === LayerType.annotation);
+
+		const features = displayedIds.reduce((array, layerId) => [...array, ...entities[layerId].data.features], []);
+		return this.showAnnotation(featureCollection(features) as FeatureCollection);
+	}
+
+	showAnnotation(annotationsLayer: FeatureCollection<any>): Observable<boolean> {
+		const annotationsLayerEntities = this.annotationsLayerToEntities(annotationsLayer);
+		this.getEntities()
+			.filter(({ id }) => !annotationsLayerEntities.some((entity) => id === entity.id))
+			.forEach(({ id }) => this.removeEntity(id));
+
+		const entitiesToAdd = annotationsLayerEntities.filter((entity) => {
+			const oldEntity = this.idToEntity.get(entity.id);
+			if (oldEntity) {
+				const isShowMeasuresDiff = oldEntity.originalEntity.showMeasures !== entity.showMeasures;
```

Comment:
check if this method and others like it can be in common imagery utils (that will be used both for cesium and ol)